docs/claudie-workflow/claudie-workflow.md

```diff
@@ -29,56 +28,57 @@
+Context box is Claudie's "control unit". It holds pending configs, which need to be processed, periodically checks for new/changed configs and receives new configs from `frontend`.
 
 ### API
-
 ```go
-  //Save config parsed by Frontend
+  // Saves the config parsed by Frontend.
   rpc SaveConfigFrontEnd(SaveConfigRequest) returns (SaveConfigResponse);
-  //Save config parsed by Scheduler
+  // Saves the config parsed by Scheduler.
   rpc SaveConfigScheduler(SaveConfigRequest) returns (SaveConfigResponse);
-  //Save config parsed by Builder
+  // Saves the config parsed by Builder.
   rpc SaveConfigBuilder(SaveConfigRequest) returns (SaveConfigResponse);
-  //Get single config from database
+  // Gets a single config from the database.
   rpc GetConfigFromDB(GetConfigFromDBRequest) returns (GetConfigFromDBResponse);
   // *(NEEDS DELETION)*
   rpc GetConfigByName(GetConfigByNameRequest) returns (GetConfigByNameResponse);
-  //Get config from scheduler queue of pending configs
+  // Gets a config from Scheduler's queue of pending configs.
   rpc GetConfigScheduler(GetConfigRequest) returns (GetConfigResponse);
-  //Get config from builder queue of pending configs
+  // Gets a config from Builder's queue of pending configs.
   rpc GetConfigBuilder(GetConfigRequest) returns (GetConfigResponse);
-  //Get all configs from database
+  // Gets all configs from the database.
   rpc GetAllConfigs(GetAllConfigsRequest) returns (GetAllConfigsResponse);
-  // Sets the manifest to null which forces the deletion of the infra,
-  // defined by the manifest, next time when the config will be picked up.
+  // Sets the manifest to null, effectively forcing the deletion of the infrastructure
+  // defined by the manifest on the very next config (diff-) check.
   rpc DeleteConfig(DeleteConfigRequest) returns (DeleteConfigResponse);
-  // Deletes config from database.
+  // Deletes the config from the database.
   rpc DeleteConfigFromDB(DeleteConfigRequest) returns (DeleteConfigResponse);
 ```
 ### Flow
-
-- Receives config file from Frontend calculates `msChecksum` and saves it to the database
-- Periodically push config where `msChecksum` != `dsChecksum` to the schedulerQueue
-- Periodically push config where `dsChecksum` != `csChecksum` to the builderQueue
-- Receives config with desiredState from Scheduler
+- Receives a `config` from Frontend, calculates its `msChecksum` and saves it to the database
+- Periodically checks for `config` changes and pushes the `config` to the `schedulerQueue` if `msChecksum` != `dsChecksum`
+- Periodically checks for `config` changes and pushes the `config` to the `builderQueue` if `dsChecksum` != `csChecksum`
+- Receives a `config` with the `desiredState` from Scheduler
 - Checks if `dsChecksum` == `msChecksum`
-    - `true` -> saves it to the database
-    - `false` -> it ignores this config
-- Receives config with currentState from Builder and saves it to the database
+  - `true` ⇒ ignores this `config`
```

Comment:
cc @MiroslavRepka, @Despire 
could you guys, please, validate this?

the description here was using a flipped logic, which made me have a look at [the relevant code](https://github.com/Berops/claudie/blob/master/services/context-box/server/cbox.go#L99) and correct the documentation (I believe).

Reply:
[the code that you refer to](https://github.com/Berops/claudie/blob/master/services/context-box/server/cbox.go#L99) actually refers to the second point `Periodically checks for config changes and pushes the config to the schedulerQueue if msChecksum != dsChecksum`

Reply:
When I look at it the check for `dsChecksum` == `msChecksum` is missing from the [function]( https://github.com/Berops/claudie/blob/master/services/context-box/server/server.go#L42-L57) I guess it was removed? @MiroslavRepka

Reply:
I vaguely remember something along those lines there a while back. I guess we removed it at some point. In any way, what @Despire wrote is correct, the correct code for that section of docs is `SaveConfig(Builder|Scheduler)()`, which is called by (Builder|Scheduler) when they are done processing particular config.

Reply:
> [the code that you refer to](https://github.com/Berops/claudie/blob/master/services/context-box/server/cbox.go#L99) actually refers to the second point `Periodically checks for config changes and pushes the config to the schedulerQueue if msChecksum != dsChecksum`

that's right, correct, thanks, my bad.


> When I look at it, the check for `dsChecksum` == `msChecksum` is missing from the [function](https://github.com/Berops/claudie/blob/master/services/context-box/server/server.go#L42-L57) I guess it was removed? @MiroslavRepka

looks like it was left up to the DB layer to handle it ([`database.UpdateDs(config)`](https://github.com/Berops/claudie/blob/master/services/context-box/server/server.go#L47)?
[`SaveConfigFrontend`](https://github.com/Berops/claudie/blob/master/services/context-box/server/server.go#L67-L69) still seems to perform the check, though.

bottom line is, I should probably remove the part about the "check" from the documentation, since there are none of the described kind. what do you think?

Reply:
Yes, I would also remove it, and just rewrite it the same as with builder, i.e. 

Receives a `config` with `currentState` from Scheduler and saves it to the database